src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function isNegative(first: number, second: number, third: number = 1): void {
+  if (first <= 0 || second <= 0 || third <= 0) {
+    throw new Error('Numbers are negative. Input positive numbers');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    isNegative(a, b, c);
+
+    const isTriangle = a + b <= c || a + c <= b || c + b <= a;
+
+    if (isTriangle) {
+      throw new Error('This is not a triangle. Input correct data');
+    }
+  }
 
+  getArea():number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(semiPerimeter * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b) * (semiPerimeter - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius is negative. Input positive radius');
+    }
```

Comment:
you're already have the function that can check that input values are bigger than 0
let's call those function for each figure